src/meson.build

```diff
@@ -2,17 +2,17 @@ nvapi_src = files([
   'util/util_string.cpp',
   'util/util_env.cpp',
   'util/util_log.cpp',
-  'sysinfo/vulkan.cpp',
-  'sysinfo/nvml.cpp',
-  'sysinfo/nvapi_output.cpp',
-  'sysinfo/nvapi_adapter.cpp',
-  'sysinfo/nvapi_adapter_registry.cpp',
+  'nvapi/vulkan.cpp',
+  'nvapi/nvml.cpp',
+  'nvapi/nvapi_output.cpp',
+  'nvapi/nvapi_adapter.cpp',
+  'nvapi/nvapi_adapter_registry.cpp',
   'resource_factory.cpp',
```

Comment:
Do you intend to share this between nvapi and nvofapi? Maybe it could be moved to `util` if that's the case (and to `nvapi/` otherwise, or just renamed to `nvapi_resource_factory.cpp`).

Reply:
Yes, for testing we need to abstract the Vulkan interaction away behind our own Vulkan wrapper, like we did within `sysinfo`. The resource factory has to create the wrapper so that we can mock this away for tests (the tests have their own resource factory which returns the mocks). I hope it works nicely to share the wrapper and resource creation between `nvapi` and `nvofapi`, but I haven't played yet.